### DISCUSSION TITLE
AWS Codeartifacts

### DISCUSSION THREAD

**Discussion by sharabhshukla** [astral-sh/rye#840] [Q&A]
I am building a python package. I would not want to upload or publish to pypi. Instead I want to publish it to AWS code articats, basically it's a private pypi server. 

I uploaded the package, using publish command and giving it the url, using rye publish.

But how do I install it. when I use, rye add --url .... it does not work the authentication fails. I can use it if I use pip install with updated sources it works.

**Comment by ischaojie** [comment]
You can config custom sources: https://rye-up.com/guide/sources/

```
[[sources]]
name = "company-internal"
url = "https://company.internal/simple/"
```

**Comment by acuitymd-filip** [comment]
Not sure if this helps, but I described a working, if somewhat hacky, solution to get something like this working with GCP's Artifact Registry here: https://github.com/astral-sh/rye/discussions/958
